app/src/main/java/com/mk/tasky/agenda/presentation/detail/task/DetailTaskViewModel.kt

```diff
@@ -0,0 +1,80 @@
+package com.mk.tasky.agenda.presentation.detail.task
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.mk.tasky.agenda.presentation.home.HomeItemOptions
+import kotlinx.coroutines.launch
+import java.time.LocalDateTime
+import javax.inject.Inject
+
+class DetailTaskViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle
+) : ViewModel() {
+    var state by mutableStateOf(DetailTaskState())
+        private set
+
+    init {
+        savedStateHandle.get<String>("date")?.let {
+            state = state.copy(
+                date = LocalDateTime.parse(it).toLocalDate(),
+                time = LocalDateTime.parse(it).toLocalTime()
+            )
+        }
+
+        val itemId = savedStateHandle.get<String>("id")
+        if (itemId != null) {
+            viewModelScope.launch {
+                /*val reminder = reminderUseCases.getReminder(itemId)
+                state = state.copy(
+                    id = itemId,
+                    title = reminder.title,
+                    description = reminder.description,
+                    date = reminder.dateTime.toLocalDate(),
+                    time = reminder.dateTime.toLocalTime(),
+                    reminder = calculateRemindAtTime(reminder)
+                )*/
```

Comment:
I'd always recommend to remove commented out code before pushing, since that just adds unnecessary noise to your code and causes everybody to question whether this can be removed or not ("so let's better keep it" 😄 ). If you need it back, version control got your back